pydantic/typing.py

```diff
@@ -70,8 +77,50 @@ def evaluate_forwardref(type_: ForwardRef, globalns: Any, localns: Any) -> Any:
             from typing_extensions import Literal
         except ImportError:
             Literal = None
+
+    def get_args(t: Type[Any]) -> Tuple[Any, ...]:
+        return getattr(t, '__args__', ())
+
+    def get_origin(t: Type[Any]) -> Optional[Type[Any]]:
+        return getattr(t, '__origin__', None)
+
+
 else:
-    from typing import Literal
+    from typing import Literal, get_args as typing_get_args, get_origin as typing_get_origin
+
+    def get_origin(tp: Type[Any]) -> Type[Any]:
+        return typing_get_origin(tp) or getattr(tp, '__origin__', None)
```

Comment:
Maybe add a comment describing why we need `getattr(tp, '__origin__', None)`

If I am not mistaken its mainly (if not all) because of the crappy code I inserted in conlist (and thus copied to conset) to deal with the fact that cython cant handle Generic inheritance

Reply:
Yes I was planning to have a deeper look on implementation of our custom types first before ;)

Reply:
@hmvp Sorry I took so long to come back on this. Thanks for the remarks and your work on this